keras_preprocessing/image.py

```diff
@@ -1438,6 +1438,10 @@ def _get_batches_of_transformed_samples(self, index_array):
                     index=j,
                     hash=np.random.randint(1e4),
                     format=self.save_format)
+
+                if not os.path.exists(self.save_to_dir):
+                    os.mkdir(self.save_to_dir)
```

Comment:
Would you like to use `os.makedirs` so that the intermediate directories are created?
https://docs.python.org/2/library/os.html#os.makedirs
https://docs.python.org/3.6/library/os.html#os.makedirs

Reply:
Thanks, @juharris! That sounds like a better approach! I'll update it.